ingestion/src/main/java/feast/store/serving/redis/RedisCustomIO.java

```diff
@@ -247,24 +257,22 @@ public void execute() {
                       }
                     });
                 pipeline.exec();
-                pipeline.sync();
+                pipeline.clear();
                 mutations.clear();
               }
 
               @Override
               public Boolean isExceptionRetriable(Exception e) {
-                return e instanceof JedisConnectionException;
+                return e instanceof RedisConnectionException ||
+                    e instanceof ExecutionException ||
+                    e instanceof InterruptedException;
               }
 
               @Override
               public void cleanUpAfterFailure() {
-                try {
-                  pipeline.close();
-                } catch (IOException e) {
-                  log.error(String.format("Error while closing pipeline: %s", e.getMessage()));
+                if (pipeline != null) {
+                  pipeline.clear();
```

Comment:
If no retries are currently set the pipeline will not be cleared, could possibly lead to some unintended behavior?  Since a new pipeline is only spawned per bundle.

Reply:
pipeline.clear() is called after exec(); line 260. (although it's not necessary as exec will conclude the transaction and wait for the result)

Reply:
If an exception occurs during exec() will pipeline.clear() still be called is what I mean

Reply:
We need to call clear() before starting a transaction in the case of retry.

Reply:
It's possible for retries to be set to 0, and if `pipeline.exec()` throws an error, `pipeline.clear()` will not be called before the next `processElement` in the batch.